diff --git a/CMake/resolve_dependency_modules/arrow/CMakeLists.txt b/CMake/resolve_dependency_modules/arrow/CMakeLists.txt
index 4f4e40319..046222487 100644
--- a/CMake/resolve_dependency_modules/arrow/CMakeLists.txt
+++ b/CMake/resolve_dependency_modules/arrow/CMakeLists.txt
@@ -94,8 +94,8 @@ if(VELOX_ENABLE_ARROW)
     CMAKE_ARGS ${ARROW_CMAKE_ARGS}
     BUILD_BYPRODUCTS ${ARROW_LIBDIR}/libarrow.a ${ARROW_LIBDIR}/libarrow_testing.a ${THRIFT_LIB}
     PATCH_COMMAND
-      git apply ${CMAKE_CURRENT_LIST_DIR}/cmake-compatibility.patch && git apply
-      ${CMAKE_CURRENT_LIST_DIR}/thrift-download.patch
+      git apply ${CMAKE_CURRENT_LIST_DIR}/thrift-download.patch &&
+      python3 ${CMAKE_CURRENT_LIST_DIR}/fix-ep-policy.py
   )
 
   add_library(arrow STATIC IMPORTED GLOBAL)
